src/utils/validIdList.js

```diff
@@ -0,0 +1,22 @@
+const validIdListRegex = /(([A-Za-z]+[\w\-:.]*)\s*)*/;
+
+// A more complex regex is used here so that
+// all IDs in an array must work. Even one incorrect
+// id in the list will cause this to return false.
+function isValidIdList(idList) {
+  if (!idList) return true;
+  const matches = idList.match(validIdListRegex);
+  return matches && matches.includes(idList);
```

Comment:
could this be `/^(([A-Za-z]+[\w\-:.]*)\s*)+$/.test(idList || '')`?

Reply:
Yes, thank you thank you. I'm no regexpert so I couldn't get it to do that. The above regex would simplify it for sure.